.github/workflows/ci.yaml

```diff
@@ -13,23 +13,38 @@ concurrency:
 jobs:
   build:
     runs-on: ubuntu-24.04
-    timeout-minutes: 5
-    
+    timeout-minutes: 15
     steps:
       - uses: actions/checkout@v4
-    
       - name: Build
         run: |
           bazel build //...
-    
       - name: Test
         run: |
           bazel test --test_output=errors //...
           
+  test-e2e:
+    runs-on: ubuntu-24.04
+    timeout-minutes: 15
+    steps:
+    - uses: actions/checkout@v4  
+    - name: Test Bzlmod Example
+      working-directory: example/bzlmod
+      run: |
+        # Run gazelle to generate BUILD files
```

Comment:
Let's extract this script into a file in that directory so it's easy to run manually without the workflow.

Please add to the top `set -o errexit -o nounset -o pipefail`.